### PR TITLE
Add Coprocessor Charter (English Version)

### PR DESCRIPTION
**Pull request by breeswish**
Translate coprocessor charter from Chinese to English.

### CODE REVIEW

**Comment by winkyao** [comment]
Please make DCO check pass.

**Comment by winkyao** [comment]
Actually, we should call it "charter", not "constitution".

**Comment by breeswish** [comment]
@winkyao I intentionally changed the Google translate result "charter" to "constitution" 🤣

**Comment by Hoverbear** [comment]
Agreed wioth @gingerkidney I think Charter is a more appropriate word. Constitution is much more 'powerful'.

File: sig/coprocessor/constitution.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Coprocessor SIG Constitution
```
Comment (winkyao) [inline sig/coprocessor/constitution.md]: s/Constitution/Charter/g

File: sig/coprocessor/constitution.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+This charter follows the TiKV Community constitution, using the role definitions and responsibilities in [SIG Governace](/GOVERNACE-zh_CN.md).
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]: This needs to be linked to the English version.

File: sig/coprocessor/constitution.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+## Scope
+
+Coprocessor SIG focuses on the TiKV project Coprocessor module, which is the module in TiKV that handles TiDB's pushdown calculations. The primary responsibility of this SIG is to discuss, plan, develop, and maintain the future development of the Coprocessor module. Only Active Contributor or higher-level community members can become members of the SIG and participate in the matter, but the decisions and projects undertaken by the SIG will be made public.
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
Coprocessor SIG focuses on the Coprocessor module of the TiKV project, which is the module in TiKV that handles TiDB's pushdown calculations. The primary responsibility of this SIG is to discuss, plan, develop, and maintain the future development of the Coprocessor module. Only Active Contributors or higher-level community members can become members of the SIG and participate in SIG activities. However, the decisions and projects undertaken by the Coprocessor SIG will be made public.
```

File: sig/coprocessor/constitution.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    - Participate in Coprocessor PR Review and quality control
+    - Valid Approve / Request Change permission on Coprocessor module PR
+
+  - Committer
+    - Have the rights and obligations of Reviewer
+    - Code quality of the overall control project
+    - Guide Contributor and Reviewer
+
+- Quit
+  - SIG members will be removed from the SIG in the following cases, but retain the corresponding Active Contributor / Reviewer / Committer identity:
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
  - SIG members will be removed from the SIG in the following cases, but retain their corresponding Active Contributor/Reviewer/Committer roles:
```

File: sig/coprocessor/constitution.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    - Complete at least two tasks with difficulty Medium or a task with difficulty High
+    - The PR contribution meets at least two points:
+      - Merge Coprocessor PR total line number exceeds 1500 lines within half a year
+      - Valid Review Coprocessor PR totals over 10
+      - Valid Review Coprocessor PR total number of lines exceeds 1000 lines
+
+- Advance role rights & obligations
+  - Reviewer
+    - Participate in project design decisions
+    - Participate in Coprocessor PR Review and quality control
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
    - Participate in Coprocessor PR reviews and the quality control process
```

File: sig/coprocessor/constitution.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    - Valid Approve / Request Change permission on Coprocessor module PR
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
    - Valid Approval/Request Change permissions on PRs of the Coprocessor module
```

File: sig/coprocessor/constitution.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+- Tech Leads additional responsibilities
+  - Questions from SIG member need to be answered within 2 business days
+  - Review code timely
+  - Publish tasks (unfinished tasks need to be reassigned if SIG member exits)
+
+
+## Promtion
+
+- Assessment & Promotion System
+  - Tech Leader evaluates team members on a monthly basis to determine whether members can be promoted to Reviewer by Active Contributor:
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
  - Tech Leaders evaluate team members on a monthly basis to determine whether they can be promoted from Active Contributor to Reviewer, based on the following aspects:
```

File: sig/coprocessor/constitution.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    - Familiar with the code base
+    - Nominated for at least 2 TiKV Committer
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
    - Nominated by at least 2 TiKV Committers
```

File: sig/coprocessor/constitution.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    - PR contribution meets any of the following:
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
    - PR contributions meet any of the following:
```

File: sig/coprocessor/constitution.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+      - More than 10 Merge Coprocessor PR
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
      - More than 10 merged Coprocessor PRs
```

File: sig/coprocessor/constitution.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+      - Merge Coprocessor PR has more than 1000 lines
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
      - Merged Coprocessor PRs involve more than 1000 lines of code
```

File: sig/coprocessor/constitution.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+      - A task with a difficulty of Medium or above has been completed
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
      - Finished at least one task with a difficulty of Medium or above

File: sig/coprocessor/constitution.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+      - Presenting design ideas and adopting them into more than 3 executable tasks
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
      - Presented design ideas and adopted them in more than 3 executable tasks
```

File: sig/coprocessor/constitution.md [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+
+  - Tech Leader and TiKV Maintainer evaluate team members on a quarterly basis to determine whether members can be promoted from Reviewer to Committer:
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
  - Tech Leaders and TiKV Maintainers evaluate team members on a quarterly basis to determine whether they can be promoted from Reviewer to Committer, based on the following aspects:

```

File: sig/coprocessor/constitution.md [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+    - Show good technical judgment
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
    - Exhibited good technical judgment
```

File: sig/coprocessor/constitution.md [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+    - Nominated for at least 2 TiKV Maintainers
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
    - Nominated by at least 2 TiKV Maintainers
```

File: sig/coprocessor/constitution.md [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+    - Complete at least two tasks with difficulty Medium or a task with difficulty High
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
    - Complete at least two tasks of Medium difficulty level or one of High difficulty level
```

File: sig/coprocessor/constitution.md [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+    - The PR contribution meets at least two points:
+      - Merge Coprocessor PR total line number exceeds 1500 lines within half a year
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
      - Merged Coprocessor PRs involve more than 1500 lines of code within half a year
```

File: sig/coprocessor/constitution.md [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+      - Valid Review Coprocessor PR totals over 10
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
      - Valid Coprocessor PR reviews total over 10
```

File: sig/coprocessor/constitution.md [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+      - Valid Review Coprocessor PR total number of lines exceeds 1000 lines
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
      - Valid Coprocessor PR reviews involve over 1000 lines of code
```

File: sig/coprocessor/constitution.md [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+
+- Advance role rights & obligations
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
-  Rights & obligations for advanced roles
```

File: sig/coprocessor/constitution.md [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+  - Reviewer
+    - Participate in project design decisions
+    - Participate in Coprocessor PR Review and quality control
+    - Valid Approve / Request Change permission on Coprocessor module PR
+
+  - Committer
+    - Have the rights and obligations of Reviewer
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
    - Inherit the rights and obligations of Reviewer
```

File: sig/coprocessor/constitution.md [hunk 23]
```diff
@@ -0,0 +1,87 @@
...
+    - Code quality of the overall control project
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
    - Ensure overall code quality of the project
```

File: sig/coprocessor/constitution.md [hunk 24]
```diff
@@ -0,0 +1,87 @@
...
+    - Guide Contributor and Reviewer
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
    - Guide Contributors and Reviewers
```

File: sig/coprocessor/constitution.md [hunk 25]
```diff
@@ -0,0 +1,87 @@
...
+
+- Quit
+  - SIG members will be removed from the SIG in the following cases, but retain the corresponding Active Contributor / Reviewer / Committer identity:
+    - as a new member did not claim the task within the specified time
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
    - As a new member, not claimed any task within the specified time
```

Comment (dcalvin) [reply sig/coprocessor/constitution.md]: specified time? why not specified here?

File: sig/coprocessor/constitution.md [hunk 26]
```diff
@@ -0,0 +1,87 @@
...
+    - Inactive for one quarter in a row
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]:
```suggestion
    - Inactive for one quarter
```

File: sig/coprocessor/constitution.md [hunk 27]
```diff
@@ -0,0 +1,87 @@
...
+
+  - Reviewer will cancel the Reviewer status and reclaim the right if one of the following conditions is met (recoverable after subsequent re-assessment):
+    - No reviews for more than one quarter. Any Coprocessor related PR
+    - More than 2 Committer thinks Reviewer is not capable or active enough
```
Comment (dcalvin) [inline sig/coprocessor/constitution.md]: Please explain these lines in Chinese...

**Comment by winkyao** [comment]
@dcalvin PTAL

**Pull request merged**